docs/arcaflow/workflows/expressions.md

```diff
@@ -28,31 +85,274 @@ $.foo.bar
+For a string example, the expression `"a" > "b"` would output `false`.
+
+### Less Than
+
+The `<` operator outputs `true` if the left side is less than the right side, and `false` otherwise. The operator requires numerical or string operands. The type must be the same for both operands.
+String operands are compared using the lexicographical order of the charset.
+
+For an integer example, the expression `3 < 3` would output the boolean `false`, and `1 < 2` would output `true`.
+For a string example, the expression `"a" < "b"` would output `true`.
+
+### Greater Than or Equal To
+
+The `>=` operator outputs `true` if the left side is greater than or equal to (not less than) the right side, and `false` otherwise. The operator requires numerical or string operands. The type must be the same for both operands.
+String operands are compared using the lexicographical order of the charset.
+
+For an integer example, the expression `3 >= 3` would output the boolean `true`, `3 >= 4` would output `false`, and `4 >= 3` would output `true`.
+
+### Less Than or Equal To
+
+The `<=` operator outputs `true` if the left side is less than or equal to (not greater than) the right side, and `false` otherwise. The operator requires numerical or string operands. The type must be the same for both operands.
+String operands are compared using the lexicographical order of the charset.
+
+For example, the expression `3 <= 3` would output the boolean `true`, `3 <= 4` would output `true`, and `4 <= 3` would output `false`.
+
+### Logical AND
+
+The `&&` operator returns `true` if both the left and right sides are `true`, and `false` otherwise. This operator requires boolean operands.
+Note: There is no short-circuiting as it's currently implemented. Both the left and right are evaluated before the comparison takes place.
+
+All cases:
+| Left    | Right   | `&&`    |
+| ------- | ------- | ------- |
+| `true`  | `true`  | `true`  |
+| `true`  | `false` | `false` |
+| `false` | `true`  | `false` |
+| `false` | `false` | `false` |
+
+### Logical OR
+
+The `||` operator returns `true` if **either or both** of the left and right sides are `true`, and `false` otherwise. This operator requires boolean operands.
+Note: There is no short-circuiting as it's currently implemented. Both the left and right are evaluated before the comparison takes place.
+
+All cases:
+| Left    | Right   | `\|\|`  |
+| ------- | ------- | ------- |
+| `true`  | `true`  | `true`  |
+| `true`  | `false` | `true`  |
+| `false` | `true`  | `true`  |
+| `false` | `false` | `false` |
+
+## Unary Operations
+
+Unary operations are operations that have one input. They are formatted as one operator to the left of the operand expression.
+
+| Operator | Description        |
+| ---------|--------------------|
+| -        | [Negation](#Negation)           |
+| !        | [Logical complement](#logical-complement) |
+
+### Negation
+
+The negation operator negates the numeric value from the input expression.
```

Comment:
I think here and in "logical complement" are the only places you refer to an operand character by *name* (dash, exclamation point) as well as by literal character. While I don't object, it's also a bit odd that it's only done here. And rather than try to do that everywhere, it might be more consistent to simply drop the names here. (Or else look up the formal UTF-8 code point names and use them consistently: e.g., "HYPHEN-MINUS (U+002D)", "EXCLAMATION MARK (U+0021)"...)

Reply:
Since I am re-clarifying that that it's before the expression, I think the extra detail here is worth it for the clarity.